File: src/main/scala/model/equipment/Armor.scala
package model.equipment

/**
  * Created by Damien on 3/12/2016.
  */
case class Armor(
                  override val name: String,
                  override val description: String,
                  override val availability: String,
                  val locations: Seq[String],
                  val armorPoints: Int,
                  val weight: String,
                  val craftsmanship: String = "Common"
                ) extends Item(name, description, availability)

File: src/main/resources/pluginresource/js/app/types/character/items/Armor.ts
import {Item, ItemType, Availability, Craftsmanship} from "./Item";
/**
 * Created by Damien on 7/9/2016.
 */
export class Armor extends Item {
    private _locations:Array<string>;
    private _ap:number;
    private _armorType:string;

    constructor(name:String, availability:Availability, locations:Array<string>, ap:number, armorType:string, weight?:Number, craftsmanship?:Craftsmanship) {
        super(name, ItemType.Armor, availability, weight, craftsmanship);
        this._locations = locations;
        this._ap = ap;
        this._armorType = armorType;
    }

    get locations():Array<string> {
        return this._locations;
    }

    get ap():number {
        return this._ap;
    }

    get armorType():string {
        return this._armorType;
    }
}